### ISSUE TITLE
bunx failed to install indirect deps

### ISSUE THREAD

**Issue by xlc** [oven-sh/bun#7593]
### What version of Bun is running?

1.0.16+800fb1290

### What platform is your computer?

Darwin 23.1.0 arm64 arm

### What steps can reproduce the bug?

`bunx @acala-network/chopsticks@0.9.3 --config kusama`

### What is the expected behavior?

With `npx @acala-network/chopsticks@0.9.3 --config kusama`, it can run the command correctly, but not for bunx

### What do you see instead?

```
$ bunx @acala-network/chopsticks@latest --config kusama
[20:20:17.811] INFO (93983): Loading config file https://raw.githubusercontent.com/AcalaNetwork/chopsticks/master/configs/kusama.yml
/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/typeorm/driver/sqlite/SqliteDriver.js:143
            throw new DriverPackageNotInstalledError_1.DriverPackageNotInstalledError("SQLite", "sqlite3");
                  ^

DriverPackageNotInstalledError: SQLite package has not been found installed. Try to install it: npm install sqlite3 --save
    at SqliteDriver.loadDependencies (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/typeorm/driver/sqlite/SqliteDriver.js:143:19)
    at new SqliteDriver (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/typeorm/driver/sqlite/SqliteDriver.js:25:14)
    at DriverFactory.create (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/typeorm/driver/DriverFactory.js:44:24)
    at new DataSource (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/typeorm/data-source/DataSource.js:53:59)
    at openDb (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/@acala-network/chopsticks-db/dist/cjs/db/index.js:55:20)
    at new SqliteDatabase (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/@acala-network/chopsticks-db/dist/cjs/index.js:30:45)
    at setupContext (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/@acala-network/chopsticks/dist/cjs/context.js:101:23)
    at setupWithServer (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/@acala-network/chopsticks/dist/cjs/setup-with-server.js:16:53)
    at Object.handler (/private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/@acala-network/chopsticks/dist/cjs/cli.js:56:39)
    at /private/tmp/@acala-network/chopsticks@latest--bunx/node_modules/yargs/build/index.cjs:1:8993

Node.js v18.17.1
```

**Comment by Jarred-Sumner** [comment]
Please run `bun upgrade --canary` and try again. This issue is fixed in the canary build of Bun, and will be included in the v1.0.17 release.

**Comment by xlc** [comment]
Looks good with canary!
